npc/ipset/ipset.go

```diff
@@ -67,3 +78,38 @@ func doExec(args ...string) error {
 	}
 	return nil
 }
+
+// Reference-counting
+type key struct {
+	ipsetName Name
+	entry     string
+}
+
+// note no locking is required as all operations are serialised in the controller
```

Comment:
*Minor*: the fact this was relatively long & painful to track is maybe a hint that we should do it differently.